scripts/Calibration/tofpd/diagnostics.py

```diff
@@ -78,3 +117,109 @@ def plot2d(workspace, tolerance: float=0.001, peakpositions: np.ndarray=DIAMOND,
+    """
+
+    ws_new = _get_difc_ws(calib_new, instr_ws)
+    if ws_new is None:
+        raise TypeError("Expected to receive a workspace or filename, got None.")
+
+    ws_old = _get_difc_ws(calib_old, instr_ws)
+    if ws_old is None:
+        # If no second workspace is given, then load default instrument to compare against
+        instr_name = ws_new.getInstrument().getName()
+        ws_old = CalculateDIFC(InputWorkspace=ws_new, OutputWorkspace="__difc_{}".format(instr_name))
+
+    delta = ws_new - ws_old
+
+    use_mask = False
+    if mask is not None and mtd.doesExist(str(mask)):
+        use_mask = True
+
+    # Plotting below taken from addie/calibration/CalibrationDiagnostics.py
+    theta_array = []
+    phi_array = []
+    value_array = []
+    masked_theta_array = []
+    masked_phi_array = []
+    info = delta.spectrumInfo()
+    for det_id in range(info.size()):
+        pos = info.position(det_id)
+        phi = np.arctan2(pos[0], pos[1])
+        theta = np.arccos(pos[2] / pos.norm())
+        if use_mask and mtd[str(mask)].dataY(det_id):
+            masked_theta_array.append(theta)
+            masked_phi_array.append(phi)
+        else:
+            theta_array.append(theta)
+            phi_array.append(phi)
+            percent = 100.0 * np.sum(delta.dataY(det_id)) / np.sum(ws_old.dataY(det_id))
+            value_array.append(percent)
+
+    # Use the largest solid angle for circle radius
+    sample_position = info.samplePosition()
+    maximum_solid_angle = 0.0
+    for det_id in range(info.size()):
+        maximum_solid_angle = max(maximum_solid_angle, delta.getDetector(det_id).solidAngle(sample_position))
+
+    # Convert to degrees for plotting
+    theta_array = np.rad2deg(theta_array)
+    phi_array = np.rad2deg(phi_array)
+    maximum_solid_angle = np.rad2deg(maximum_solid_angle)
+    if use_mask:
+        masked_phi_array = np.rad2deg(masked_phi_array)
+        masked_theta_array = np.rad2deg(masked_theta_array)
+
+    # Radius also includes a fudge factor to improve plotting.
+    # May need to add finer adjustments on a per-instrument basis.
+    # Small circles seem to alias less than rectangles.
+    radius = maximum_solid_angle * 8.0
+    patches = []
+    for x1, y1 in zip(theta_array, phi_array):
+        circle = Circle((x1, y1), radius)
+        patches.append(circle)
+
+    masked_patches = []
+    for x1, y1 in zip(masked_theta_array, masked_phi_array):
+        circle = Circle((x1, y1), radius)
+        masked_patches.append(circle)
+
+    # Matplotlib requires this to be a Numpy array.
+    colors = np.array(value_array)
+    p = PatchCollection(patches)
+    p.set_array(colors)
+    p.set_clim(vrange[0], vrange[1])
+    p.set_edgecolor('face')
+
+    fig, ax = plt.subplots()
+    ax.add_collection(p)
+    mp = PatchCollection(masked_patches)
+    mp.set_facecolor('gray')
+    mp.set_edgecolor('face')
+    ax.add_collection(mp)
+    cb = fig.colorbar(p, ax=ax)
+    cb.set_label('delta DIFC (%)')
```

Comment:
label should be "DIFC similarity." It is not a percent because you aren't multiplying by "100"